lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3965,6 +3965,9 @@ void TypeChecker::validateDecl(ValueDecl *D) {
     // as it has a valid interface type.
     FD->setSignatureIsValidated();
 
+    // TODO(TF-789): Figure out the proper way to typecheck these.
+    checkDeclDifferentiableAttributes(FD);
```

Comment:
This works for now, but keep in mind validateDecl() is going away soon, so you'll want to request-ify this sooner rather than later.